design/open/lakefs-on-kv-testing-plan.md

```diff
@@ -0,0 +1,133 @@
+# lakeFS on KV - Testing Plan
+
+## Global Testing
+
+### Migration
+* Global tests to verify and benchmark the migration process
+* Need to verify that the system behavior is not affected by the move to the KV DB
+* [Question] - Do we need to dispose of these once migration to KV is globally completed?
+  * Yes. Migration code (and related tests) should be removed from the code, once the entire deployment of KV Store is done
+
+#### Operational Testing
+* Set of tests that will run DB migration as part of the test
+* Currently, system tests assume lakeFS is up and running. Need to add ability to perform migration in the middle of the test (stop lakeFS -> run&verify migration -> start lakeFS ?)
+
+#### Data Level Testing
+**Note:** This section is on a "Nice to Have" basis, as it is pretty much covered by the previous section
+* ~~Dump~~ Extract (all) data before and after migration and compare it
+* Verifies data is preserved through migration, and is readable by the KV Store
+* Need to write appropriate dumpers. Can be done per package, as we go
+* Can be done both on DB package level, to verify the package itself, and on the using packages level, to verify data usage is not affected
+* [Question] - How do we define a covering data set, to test? How do we create
+
+### Performance
+* **A benchmark for the migration process itself:**
+  * How do we load the DB with data to migrate for benchmarking?
+  * Ideally, run on several data sizes (tens / thousands / millions branches & commits, various numbers of users, etc.) to get sense of how scaling will affect
+  * What is the expected/acceptable outcome?
+  * Should be built gradually, as we support and migrate additional package
+
+* **Benchmark comparison of both DBs**
```

Comment:
We might want to benchmark the migration process itself to see that it does not suffer from serious performance issues once deployed in the field with costumers who have a lot of data to migrate (resulting in an unacceptably long downtime)

Reply:
Absolutely correct. It is coevered in the previous section